osconfig_tests/osconfig_server/osconfig_service_wrapper.go

```diff
@@ -0,0 +1,122 @@
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package osconfigserver contains wrapper around osconfig service APIs and helper methods
+package osconfigserver
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/config"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/junitxml"
+	"github.com/kylelemons/godebug/pretty"
+	"google.golang.org/api/option"
+
+	osconfig "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
+	osconfigpb "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
+)
+
+var dump = &pretty.Config{IncludeUnexported: true}
+var osconfigClient *osconfig.Client
+
+func getOsConfigClient(ctx context.Context, logger *log.Logger) (*osconfig.Client, error) {
+
+	if osconfigClient != nil {
+		return osconfigClient, nil
+	}
+
+	client, err := osconfig.NewClient(ctx, option.WithEndpoint(config.SvcEndpoint()), option.WithCredentialsFile(config.OAuthPath()))
+
+	if err != nil {
+		logger.Printf("error while creating osconfig client: %s\n", err)
+	}
+	return client, err
+}
+
+// CreateOsConfig is a wrapper around createOsConfig API
+func CreateOsConfig(ctx context.Context, logger *log.Logger, req *osconfigpb.CreateOsConfigRequest) (*osconfigpb.OsConfig, error) {
+	client, err := getOsConfigClient(ctx, logger)
+
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Printf("create osconfig request:\n%s\n\n", dump.Sprint(req))
+
+	res, err := client.CreateOsConfig(ctx, req)
+	if err != nil {
+		logger.Printf("error while creating osconfig:\n%s\n\n", err)
+		return nil, err
+	}
+	logger.Printf("create osconfig response:\n%s\n\n", dump.Sprint(res))
+
+	return res, nil
+}
+
+// ListOsConfigs is a wrapper around listOsConfigs API
+func ListOsConfigs(ctx context.Context, logger *log.Logger, req *osconfigpb.ListOsConfigsRequest) *osconfig.OsConfigIterator {
+	client, err := getOsConfigClient(ctx, logger)
+
+	if err != nil {
+		return nil
+	}
+
+	logger.Printf("List osconfig request:\n%s\n\n", dump.Sprint(req))
+
+	resp := client.ListOsConfigs(ctx, req)
+	if resp == nil {
+		logger.Printf("error while listing osconfig:\n%v\n\n", resp)
+		return nil
+	}
+
+	return resp
+}
+
+// DeleteOsConfig is a wrapper around deleteOsConfig API
+func DeleteOsConfig(ctx context.Context, logger *log.Logger, req *osconfigpb.DeleteOsConfigRequest) error {
```

Comment:
Can you just combine these two functions into one called DeleteOsConfig, no reason to have to at this surface

Reply:
my idea of having 2 separate functions was that DeleteOsConfig will be used to imitate user behavior where user at certain step of his workflow deletes an osconfig.
Whereas Cleanup is supposed to be a part of test framework which will be used at the end to remove the left over osconfigs which was created by the user. Deleting this osconfig is not supposed be a part of this test. A noob example of such test case is, user creates an osconfig with a specific name and then tries to create another osconfig with same name and he should be able to do so. In this case, cleaning up the actual osconfig after the test completes is not a part of actual test.
I think, i should move the CleanupOsConfig method to a test framework related package.

Reply:
if you wrap the config like I do for instances you can create a cleanup hook:
https://github.com/GoogleCloudPlatform/compute-image-tools/blob/master/osconfig_tests/compute/instance.go#L28

Reply:
sure, yeah. I can do that. thanks